commands/message_daemon_test.go

```diff
@@ -104,7 +105,9 @@ func TestMessageWait(t *testing.T) {
 
 		d.RunSuccess("mining once")
 
-		wg.Wait()
+		if !th.WaitWithTimeout(&wg, time.Second*5) {
```

Comment:
Would it make sense to push these deeper so that *all* the call sites don't need this? Like if every method on `d` carried a timeout, configured at construction?